### PR TITLE
dialects: remove some stencil dead code.

### PR DESCRIPTION
**Pull request by PapyChacal**

### CODE REVIEW

File: xdsl/dialects/stencil.py [hunk 1]
```diff
@@ -628,13 +622,6 @@ def get_rank(self) -> int:
             assert self.bounds is not None
             return len(self.bounds.lb)
 
-    def get_num_results(self) -> int:
```
Comment (n-io) [inline xdsl/dialects/stencil.py:631]: This looks useful, why remove?

**Comment by superlopuh** [comment]
What's the latest on this?

**Comment by superlopuh** [comment]
Let's merge or close this? @AntonLydike @PapyChacal @n-io @dk949

**Comment by PapyChacal** [comment]
I don't mind either way, whatever suits the current devs!

**Comment by n-io** [comment]
Happy to see this merged.

**Review by n-io — APPROVED** [review]

**Pull request merged**